portfolio/src/main/webapp/script.js

```diff
@@ -16,44 +16,69 @@ window.onload = function() {
 
     var commentCapacity = document.getElementById("comment-number").value;
     
     fetch("/data?comment-capacity=" + commentCapacity).then(response => response.json()).then((commentsJson) => {
 
-        console.log(commentsJson);
         const commentSectionElement= document.getElementById("comments");
 
         for (comment of commentsJson) {
-            commentSectionElement.appendChild(createCommentElement(comment));
+            console.log(comment);
+            json = JSON.parse(comment);
+            commentSectionElement.appendChild(createcommentElementFromJson(json));
         }
     });
 }
 
-function createCommentElement(text) {
-    const pElement = document.createElement("p");
-    pElement.innerText = text;
-    return pElement;
+function createcommentElementFromJson(json) {
+    
+    var nameElement = document.createElement("b");
+    var nameText = document.createTextNode(json.name + ": ");
+    nameElement.appendChild(nameText);
+    var commentText = document.createTextNode(json.comment);
+
+    var comment = document.createElement("p");
+    comment.appendChild(nameElement);
+    comment.appendChild(commentText);
+    return comment;
 }
 
 function deleteComments() {
-    console.log("fetching deletion");
     fetch("/delete-data", { method: "post" }).then(response => console.log(response.text()));
     getComments();
 }
 
 function printCommentSender() {
-    fetch("/Authenticate").then(userData => userData.json()).then(nicknameJson => {
-        console.log(nicknameJson);
+    fetch("/user-data").then(userData => userData.json()).then(Json => {
         commentSenderDiv = document.getElementById("comment-sender-div");
-        commentSenderDiv.innerHTML =
-        `<p>Posting as <b>${nicknameJson.nickname}</b>. visit <a class="hyperlink" href="/settings">Settings</a> to update nickname.</p>
-        <form id="comment-box" action="/data" method="POST">
-            <textarea name="comment-input" placeholder="Comment on this page"></textarea>
-            <input value="Send" type="submit">
-        </form>`;
+
+        if (Json.userLoggedIn) {
+            commentSenderDiv.innerHTML = commentSender(Json.nickname);
+            document.getElementById("nickname-space").innerText = Json.nickname;
+        } else {
+            commentSenderDiv.innerHTML = LoginLink(Json.loginUrl);
+        }
+        
     });
+}
+
+function LoginLink(loginUrl) {
+    linkHtml =
+    `<p><a class="hotpink" href="${loginUrl}">Login</a> to post a comment.</p>`;
+    return linkHtml;
+}
+
+function commentSender() {
+    senderHtml = 
+    `<p>Commenting as <span class="hotpink"><b id="nickname-space"></b></span>. Visit <a class="hotpink" href="/settings.html">Settings</a> to update nickname.</p>
+    <form id="comment-box" action="/data" method="POST">
+        <textarea name="comment-input" placeholder="Comment on this page"></textarea>
+        <input value="Send" type="submit">
+    </form>`;
```

Comment:
Building the HTML like this seems hard to read/maintain. Can you do this using Javascript instead?

Also, I may be missing something, but it looks like above you call this function with a parameter, but here it doesn't have a parameter (and this part doesn't seem like it includes the nickname)

Reply:
The nickname insertion now happens after this function is called. I did this to avoid allowing users to inject html with their nickname, I had forgotten to remove the parameter.

Also I tried building the html using Javascript functions, it was a lot longer to write and much less readable. I'm interested in hearing more of your advice on building html in Javascript. I'll probably ask in today's pod sync.

Reply:
Hm, yeah I see what you mean. Maybe instead of building it here, you could include it in your HTML, with visibility:hidden at first. And then here in the javascript, you could simply set it to be visible.

Happy to chat about it in the pod sync as well, maybe others will have some good ideas here as well.